### ISSUE TITLE
Azuracast stop working since the last update

### ISSUE THREAD

**Issue by arusi01** [AzuraCast/AzuraCast#2811]
Azuracast stop working since the last update 
AzuraCast • v0.10.2, #1229f2b (2020-05-07 0:05) • Docker • PHP 7.4

when I hit the next song button I get this error

Telnet failure: Connection refused (111)

and Shoutcast Streams Status has Available Streams: 0

Liquidsoap Log -> Fatal error: out of memory

**Comment by SlvrEagle23** [comment]
@arusi01 I'm not sure what would be causing that, but it seems Liquidsoap may have crashed because you don't have enough system RAM available.

One thing that can be helpful, especially for the smaller VPSes that don't offer a lot of RAM, would be to create a swap space: https://www.digitalocean.com/community/tutorials/how-to-add-swap-space-on-ubuntu-18-04

**Comment by davidkamp** [comment]
I have the same issue, and the stats of my server don't come anywhere near their max....
Is this being investigated?

**Comment by arusi01** [comment]
My server used memory:

free -m
              total        used        free      shared  buff/cache   available
Mem:           7978        1681        4305          40        1991        5961
Swap:             0           0           0

**Comment by arusi01** [comment]
Updated to Powered by AzuraCast • v0.10.2, #5102ce8 (2020-05-10 7:34) • Docker • PHP 7.4

and I still have the problem

**Comment by arusi01** [comment]
I temporarily fixed it by disabling some playlists with no songs assigned.

**Comment by davidkamp** [comment]
This work around does not work here.

**Comment by davidkamp** [comment]
Here is what solved it for me: 
-Creating a new station with exactly the same files and  settings (station profile, playlist, autodj etc)  as the one I had before the update.
This newly created one works, the station created on the previous azuracast did not work no matter the settings.

Also it seems the crossfade duration now works as expected which is great!

**Comment by arusi01** [comment]
Creating a new station with exactly the same files and settings does not work here.

**Comment by arusi01** [comment]
![Sin título-1](https://user-images.githubusercontent.com/51458416/82840536-59b64300-9ea9-11ea-9ce5-36f85a35de3c.jpg)

**Comment by arusi01** [comment]
# Welcome to the AzuraCast Liquidsoap configuration editor.
# Using this page, you can customize several sections of the Liquidsoap configuration.
# The non-editable sections are automatically generated by AzuraCast.

set("init.daemon", false)
set("init.daemon.pidfile.path","/var/azuracast/stations/radio/config/liquidsoap.pid")
set("log.stdout", true)
set("log.file", false)
set("server.telnet",true)
set("server.telnet.bind_addr","0.0.0.0")
set("server.telnet.port", 8304)
set("harbor.bind_addrs",["0.0.0.0"])

set("tag.encodings",["UTF-8","ISO-8859-1"])
set("encoder.encoder.export",["artist","title","album","song"])

setenv("TZ", "America/Argentina/Buenos_Aires")

playlist_bandas_con_historia_1 = playlist(id="playlist_bandas_con_historia_1",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_bandas_con_historia_1.m3u")
playlist_bandas_con_historia_1 = audio_to_stereo(id="stereo_playlist_bandas_con_historia_1", playlist_bandas_con_historia_1)
playlist_bandas_con_historia_1 = cue_cut(id="cue_playlist_bandas_con_historia_1", playlist_bandas_con_historia_1)

playlist_himno = playlist(id="playlist_himno",mode="normal",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_himno.m3u")
playlist_himno = audio_to_stereo(id="stereo_playlist_himno", playlist_himno)
playlist_himno = cue_cut(id="cue_playlist_himno", playlist_himno)

playlist_diario = playlist(id="playlist_diario",mode="random",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_diario.m3u")
playlist_diario = audio_to_stereo(id="stereo_playlist_diario", playlist_diario)
playlist_diario = cue_cut(id="cue_playlist_diario", playlist_diario)

playlist_bandas_con_historia_2 = playlist(id="playlist_bandas_con_historia_2",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_bandas_con_historia_2.m3u")
playlist_bandas_con_historia_2 = audio_to_stereo(id="stereo_playlist_bandas_con_historia_2", playlist_bandas_con_historia_2)
playlist_bandas_con_historia_2 = cue_cut(id="cue_playlist_bandas_con_historia_2", playlist_bandas_con_historia_2)

playlist_los_sonidos_del_silencio_jueves = playlist(id="playlist_los_sonidos_del_silencio_jueves",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_los_sonidos_del_silencio_jueves.m3u")
playlist_los_sonidos_del_silencio_jueves = audio_to_stereo(id="stereo_playlist_los_sonidos_del_silencio_jueves", playlist_los_sonidos_del_silencio_jueves)
playlist_los_sonidos_del_silencio_jueves = cue_cut(id="cue_playlist_los_sonidos_del_silencio_jueves", playlist_los_sonidos_del_silencio_jueves)

playlist_los_sonidos_del_silencio_lunes = playlist(id="playlist_los_sonidos_del_silencio_lunes",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_los_sonidos_del_silencio_lunes.m3u")
playlist_los_sonidos_del_silencio_lunes = audio_to_stereo(id="stereo_playlist_los_sonidos_del_silencio_lunes", playlist_los_sonidos_del_silencio_lunes)
playlist_los_sonidos_del_silencio_lunes = cue_cut(id="cue_playlist_los_sonidos_del_silencio_lunes", playlist_los_sonidos_del_silencio_lunes)

playlist_los_sonidos_del_silencio_martes = playlist(id="playlist_los_sonidos_del_silencio_martes",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_los_sonidos_del_silencio_martes.m3u")
playlist_los_sonidos_del_silencio_martes = audio_to_stereo(id="stereo_playlist_los_sonidos_del_silencio_martes", playlist_los_sonidos_del_silencio_martes)
playlist_los_sonidos_del_silencio_martes = cue_cut(id="cue_playlist_los_sonidos_del_silencio_martes", playlist_los_sonidos_del_silencio_martes)

playlist_los_sonidos_del_silencio_miercoles = playlist(id="playlist_los_sonidos_del_silencio_miercoles",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_los_sonidos_del_silencio_miercoles.m3u")
playlist_los_sonidos_del_silencio_miercoles = audio_to_stereo(id="stereo_playlist_los_sonidos_del_silencio_miercoles", playlist_los_sonidos_del_silencio_miercoles)
playlist_los_sonidos_del_silencio_miercoles = cue_cut(id="cue_playlist_los_sonidos_del_silencio_miercoles", playlist_los_sonidos_del_silencio_miercoles)

playlist_los_sonidos_del_silencio_viernes = playlist(id="playlist_los_sonidos_del_silencio_viernes",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_los_sonidos_del_silencio_viernes.m3u")
playlist_los_sonidos_del_silencio_viernes = audio_to_stereo(id="stereo_playlist_los_sonidos_del_silencio_viernes", playlist_los_sonidos_del_silencio_viernes)
playlist_los_sonidos_del_silencio_viernes = cue_cut(id="cue_playlist_los_sonidos_del_silencio_viernes", playlist_los_sonidos_del_silencio_viernes)

playlist_made_in_argentina_lunes = playlist(id="playlist_made_in_argentina_lunes",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_made_in_argentina_lunes.m3u")
playlist_made_in_argentina_lunes = audio_to_stereo(id="stereo_playlist_made_in_argentina_lunes", playlist_made_in_argentina_lunes)
playlist_made_in_argentina_lunes = cue_cut(id="cue_playlist_made_in_argentina_lunes", playlist_made_in_argentina_lunes)

playlist_made_in_argentina_martes = playlist(id="playlist_made_in_argentina_martes",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_made_in_argentina_martes.m3u")
playlist_made_in_argentina_martes = audio_to_stereo(id="stereo_playlist_made_in_argentina_martes", playlist_made_in_argentina_martes)
playlist_made_in_argentina_martes = cue_cut(id="cue_playlist_made_in_argentina_martes", playlist_made_in_argentina_martes)

playlist_made_in_argentina_miercoles = playlist(id="playlist_made_in_argentina_miercoles",mode="randomize",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_made_in_argentina_miercoles.m3u")
playlist_made_in_argentina_miercoles = audio_to_stereo(id="stereo_playlist_made_in_argentina_miercoles", playlist_made_in_argentina_miercoles)
playlist_made_in_argentina_miercoles = cue_cut(id="cue_playlist_made_in_argentina_miercoles", playlist_made_in_argentina_miercoles)

playlist_anuncios = playlist(id="playlist_anuncios",mode="random",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_anuncios.m3u")
playlist_anuncios = audio_to_stereo(id="stereo_playlist_anuncios", playlist_anuncios)
playlist_anuncios = cue_cut(id="cue_playlist_anuncios", playlist_anuncios)
playlist_anuncios = drop_metadata(playlist_anuncios)

playlist_jingles = playlist(id="playlist_jingles",mode="random",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/radio/playlists/playlist_jingles.m3u")
playlist_jingles = audio_to_stereo(id="stereo_playlist_jingles", playlist_jingles)
playlist_jingles = cue_cut(id="cue_playlist_jingles", playlist_jingles)
playlist_jingles = drop_metadata(playlist_jingles)

# Standard Playlists
radio = random(id="radio_standard_playlists", weights=[3], [playlist_diario])

# Once per x Songs Playlists
radio = rotate(weights=[1,8], [playlist_anuncios, radio])
radio = rotate(weights=[1,3], [playlist_jingles, radio])

# AutoDJ Next Song Script
def azuracast_next_song() =
    uri = list.hd(get_process_lines("curl -s --request POST --url http://web/api/internal/8/nextsong --form api_auth="^string.quote("87c79eef7fdf6a820158941a4cdfa3c4056b646444ccb78a5cfe3ecb3ee70491152d650fa85616d47c67477040f1b0a3e089")^""), default="")
    log("AzuraCast Raw Response: #{uri}")
    
    if uri == "" or string.match(pattern="Error", uri) then
        []
    else 
        req = request.create(uri)
        [req]                
    end
end

dynamic = request.dynamic.list(id="radio_next_song", timeout=20., retry_delay=3., azuracast_next_song)
dynamic = audio_to_stereo(id="radio_stereo_next_song", dynamic)
dynamic = cue_cut(id="radio_cue_next_song", dynamic)
radio = fallback(id="radio_autodj_fallback", track_sensitive = true, [dynamic, radio])

# Interrupting Schedule Switches
radio = switch(id="radio_interrupt_switch", track_sensitive=false, [ ({ 7h0m-8h0m }, once(playlist_bandas_con_historia_1)), ({ 12h0m-13h0m }, once(playlist_bandas_con_historia_1)), ({ 15h0m-16h0m }, once(playlist_bandas_con_historia_1)), ({ 0h0m-0h5m }, once(playlist_himno)), ({ 1h0m-2h0m }, once(playlist_bandas_con_historia_2)), ({ 18h0m-19h0m }, once(playlist_bandas_con_historia_2)), ({ 22h0m-23h0m }, once(playlist_bandas_con_historia_2)), ({ (4w) and 3h0m-4h0m }, once(playlist_los_sonidos_del_silencio_jueves)), ({ (4w) and 16h0m-17h0m }, once(playlist_los_sonidos_del_silencio_jueves)), ({ ((4w) and 23h0m-23h59m59s) or ((5w) and 00h00m-0h0m) }, once(playlist_los_sonidos_del_silencio_jueves)), ({ (4w) and 11h0m-12h0m }, once(playlist_los_sonidos_del_silencio_jueves)), ({ (1w) and 3h0m-4h0m }, once(playlist_los_sonidos_del_silencio_lunes)), ({ (1w) and 11h0m-12h0m }, once(playlist_los_sonidos_del_silencio_lunes)), ({ (1w) and 16h0m-17h0m }, once(playlist_los_sonidos_del_silencio_lunes)), ({ ((1w) and 23h0m-23h59m59s) or ((2w) and 00h00m-0h0m) }, once(playlist_los_sonidos_del_silencio_lunes)), ({ (2w) and 3h0m-4h0m }, once(playlist_los_sonidos_del_silencio_martes)), ({ (2w) and 11h0m-12h0m }, once(playlist_los_sonidos_del_silencio_martes)), ({ (2w) and 16h0m-17h0m }, once(playlist_los_sonidos_del_silencio_martes)), ({ ((2w) and 23h0m-23h59m59s) or ((3w) and 00h00m-0h0m) }, once(playlist_los_sonidos_del_silencio_martes)), ({ (3w) and 3h0m-4h0m }, once(playlist_los_sonidos_del_silencio_miercoles)), ({ (3w) and 11h0m-12h0m }, once(playlist_los_sonidos_del_silencio_miercoles)), ({ (3w) and 16h0m-17h0m }, once(playlist_los_sonidos_del_silencio_miercoles)), ({ ((3w) and 23h0m-23h59m59s) or ((4w) and 00h00m-0h0m) }, once(playlist_los_sonidos_del_silencio_miercoles)), ({ (5w) and 3h0m-4h0m }, once(playlist_los_sonidos_del_silencio_viernes)), ({ (5w) and 11h0m-12h0m }, once(playlist_los_sonidos_del_silencio_viernes)), ({ (5w) and 16h0m-17h0m }, once(playlist_los_sonidos_del_silencio_viernes)), ({ ((5w) and 23h0m-23h59m59s) or ((6w) and 00h00m-0h0m) }, once(playlist_los_sonidos_del_silencio_viernes)), ({ (1w) and 2h0m-3h0m }, once(playlist_made_in_argentina_lunes)), ({ (1w) and 10h0m-11h0m }, once(playlist_made_in_argentina_lunes)), ({ (1w) and 14h0m-15h0m }, once(playlist_made_in_argentina_lunes)), ({ (1w) and 21h0m-22h0m }, once(playlist_made_in_argentina_lunes)), ({ (2w) and 2h0m-3h0m }, once(playlist_made_in_argentina_martes)), ({ (2w) and 10h0m-11h0m }, once(playlist_made_in_argentina_martes)), ({ (2w) and 14h0m-15h0m }, once(playlist_made_in_argentina_martes)), ({ (2w) and 21h0m-22h0m }, once(playlist_made_in_argentina_martes)), ({ (3w) and 2h0m-3h0m }, once(playlist_made_in_argentina_miercoles)), ({ (3w) and 10h0m-11h0m }, once(playlist_made_in_argentina_miercoles)), ({ (3w) and 14h0m-15h0m }, once(playlist_made_in_argentina_miercoles)), ({ (3w) and 21h0m-22h0m }, once(playlist_made_in_argentina_miercoles)), ({true}, radio) ])

requests = request.queue(id="radio_requests")
requests = audio_to_stereo(id="radio_stereo_requests", requests)
requests = cue_cut(id="radio_cue_requests", requests)
radio = fallback(id="radio_requests_fallback", track_sensitive = true, [requests, radio])

add_skip_command(radio)

radio = crossfade(smart=false, duration=1.50,fade_out=1.,fade_in=1.,radio)

radio = fallback(id="radio_safe_fallback", track_sensitive = false, [radio, single(id="error_jingle", "/usr/local/share/icecast/web/error.mp3")])

# Allow for Telnet-driven insertion of custom metadata.
radio = server.insert_metadata(id="custom_metadata", radio)

# Apply amplification metadata (if supplied)
radio = amplify(override="liq_amplify", 1., radio)

# Send metadata changes back to AzuraCast
def metadata_updated(m) =
    if (m["song_id"] != "") then
        ret = list.hd(get_process_lines("curl -s --request POST --url http://web/api/internal/8/feedback --form song="^string.quote(m["song_id"])^" --form media="^string.quote(m["media_id"])^" --form playlist="^string.quote(m["playlist_id"])^" --form api_auth="^string.quote("87c79eef7fdf6a820158941a4cdfa3c4056b646444ccb78a5cfe3ecb3ee70491152d650fa85616d47c67477040f1b0a3e089")^""), default="")
        log("AzuraCast Feedback Response: #{ret}")
    end
end

radio = on_metadata(metadata_updated,radio)

# Local Broadcasts
output.icecast(%mp3(samplerate=44100, stereo=true, bitrate=128, id3v2=true), id="radio_local_1", host = "127.0.0.1", port = 8300, password = "fjdskjfhksdjh", icy_id = 1, name = "Radio", description = "", genre = "Variado, Noticias", url = "https://www.radio.com/radio", public = false, encoding = "UTF-8", protocol="icy", radio)

# Remote Relays

**Comment by arusi01** [comment]
[AzuraCast Liquidsoap configuration.txt](https://github.com/AzuraCast/AzuraCast/files/4678704/AzuraCast.Liquidsoap.configuration.txt)

**Comment by arusi01** [comment]
![Sin título-3](https://user-images.githubusercontent.com/51458416/82841351-14dfdb80-9eac-11ea-9713-38766d2c85ad.jpg)

**Comment by arusi01** [comment]
![22222](https://user-images.githubusercontent.com/51458416/82841373-304ae680-9eac-11ea-9e71-62b4f69ac6ae.jpg)

**Comment by SlvrEagle23** [comment]
@arusi01 What does your Liquidsoap configuration say? If you're getting connection refused via Telnet, that means our system can't connect to your Liquidsoap installation, which likely means it failed to launch and isn't running.

**Comment by arusi01** [comment]
[liquidsoaplog.txt](https://github.com/AzuraCast/AzuraCast/files/4679166/liquidsoaplog.txt)

**Comment by arusi01** [comment]
.

**Comment by arusi01** [comment]
[liquidsoaplog.txt](https://github.com/AzuraCast/AzuraCast/files/4679169/liquidsoaplog.txt)

**Comment by arusi01** [comment]
Where can I confirm liquidsoap is running?

![service](https://user-images.githubusercontent.com/51458416/82847886-05ba5700-9ec7-11ea-8533-bffa5b3dae39.jpg)

**Comment by arusi01** [comment]
I think the error is related to de playlist settings, try to recreate the error by adding 14 lists with the "Interrupt other songs to play at scheduled time" and "Only play one track at scheduled time." both checked and scheduled three times per day each

**Comment by SlvrEagle23** [comment]
@arusi01 I suspect this is an issue with Liquidsoap itself. You should post on [their issues section](https://github.com/savonet/liquidsoap/issues) with your Liquidsoap error log and your generated configuration to see if they can help further.

It's also possible your system is out of memory in general, and doing something like adding swap space may assist in the issue.

**Comment by arusi01** [comment]
Did you try to add the playlists? did not give any error?

**Comment by SlvrEagle23** [comment]
@arusi01 Even if I can reproduce the error, there's nothing I can do about an "out of memory" error with Liquidsoap itself. That's why I'm trying to guide you to the correct location for fixing the issue.

**Comment by arusi01** [comment]
I would like to know if it gives you the same error

**Comment by arusi01** [comment]
The fix has been identified here: https://github.com/savonet/liquidsoap/pull/1272 as an exponential increase in the list of sub-clocks associated with a given unknown clock.

**Comment by SlvrEagle23** [comment]
@arusi01 AzuraCast has finished building an updated stations image with this update pulled from the 1.4.3 development branch of Liquidsoap. You can now update your installations to take advantage of the updated code.

**Comment by arusi01** [comment]
thanks! I can confirm that the fix is working with my installation

**Comment by davidkamp** [comment]
great! what is the command to update the docker installation to the
latest version?

**Comment by SlvrEagle23** [comment]
@davidkamp Basically,

```bash
cd /var/azuracast
./docker.sh update-self
./docker.sh update
```

**Comment by dadadadaka** [comment]
Thanks! 
I updated - and now the station works when playing on the azuracast site (radio.studiokamp.com) 
It does however not play on my embedded players here: https://www.studiokamp.com/rainforest-radio/

After the update I got a message in chrome that the Letsencrypt certificate has issues and I can only login to my station via safari, not chrome. 

I then renewed the certificate following these instructions:
https://www.azuracast.com/extending/letsencrypt.html#important-considerations

No error messages. 

But my embedded players still wont play. 
They are embedded like this on my https site:

<audio src="https://radio.studiokamp.com/radio/8010/radio.mp3?1593608651" autoplay="false" controls="true" volume="1.0"></audio> 

Any ideas what is going on?
